contracts/randdrop/src/msg.rs

```diff
@@ -0,0 +1,101 @@
+use cosmwasm_schema::{cw_serde, QueryResponses};
+use cosmwasm_std::{HexBinary, Uint128};
+use nois::NoisCallback;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    /// manager if none set to info.sender.
+    pub manager: String,
+    /// Address of the Nois proxy contract
+    pub nois_proxy_address: String,
+    /// Nois proxy prices denom
+    pub nois_proxy_denom: String,
+    /// Nois proxy prices amount
+    pub nois_proxy_amount: Uint128,
+    /// Randdrop denom
+    pub randdrop_denom: String,
+}
+
+#[cw_serde]
+pub enum ExecuteMsg {
+    UpdateConfig {
+        manager: Option<String>,
+        nois_proxy_denom: Option<String>,
+        nois_proxy_amount: Option<Uint128>,
+        nois_proxy_address: Option<String>,
+        randdrop_denom: Option<String>,
+    },
+    RegisterMerkleRoot {
+        /// MerkleRoot is hex-encoded merkle root.
+        merkle_root: HexBinary,
+    },
```

Comment:
Why is this not part of instantiation? Then we don't need all the handling to ensure it is only set once.